### PR TITLE
Improvements to election settings handling plus, you know, rewriting lots of stuff

### PR DESCRIPTION
**Pull request by st421**
## Description
We need to be able to create a mapping from race IDs to last updated timestamps using [Clarity's electionsettings.json](https://results.enr.clarityelections.com/GA/105369/270998/json/en/electionsettings.json). This PR adds a converter for such settings and does 50 other things because why not! Just kidding; I know I have a problem. I will update with the list of changes soon, just wanted to get this up before Anthony goes on vacation (sorry @anthonyjpesce ☹️ ).

## Test Steps
Basically try all the examples in the readme and make sure they still work! The new settings functionality in particular requires that you run `elexclarity 105369 GA --outputType=settings --officeID=P --countyMapping=<full actual county mapping>`.

### CODE REVIEW

File: src/elexclarity/formatters/base.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        slug = slugify(name, separator="_")
+        if not self.county_lookup:  # No mapping provided
+            return slug
+        return self.county_lookup[name]
+
+    @classmethod
+    def get_timestamp(cls, input_timestamp):
+        # convert the timestamp and make sure we're in EST
+        est = tz.gettz("America/New_York")
+        return parser.parse(input_timestamp, tzinfos={"EST": est}).astimezone(est)
```
Comment (anthonyjpesce) [inline src/elexclarity/formatters/base.py:41]: Swapping `.astimezone(est)` should be all we need for the UTC thing (I had a vague memory of us using EST in Dynamo but who knows)

**Review by anthonyjpesce — APPROVED** [review]
The refactor LGTM! We def needed some additional organization. The ID stuff makes sense too (though it is pretty specific to our team if someone else wants to use this).

File: README.rst [hunk 2]
```diff
@@ -13,14 +13,14 @@ Examples:
...
 Pulling formatted data from Clarity:
 
 Pass in county mapping as a JSON object.
 Sample formats can be found in the ``tests/fixtures`` folder `here <https://github.com/washingtonpost/elex-clarity/tree/develop/tests/fixtures>`_.
 
-* ``elexclarity 105369 GA --level=precinct --countyMapping='{"Worth": "13321"}``
+* ``elexclarity 105369 GA --level=precinct --countyMapping='{"Worth": "13321"}'``
 * ``elexclarity 105369 GA --level=precinct --countyMapping='<mapping json>'``
 * ``elexclarity 105369 GA --level=county --countyMapping='<mapping json>'``
 * ``elexclarity 106210 WV --level=county --countyMapping='<mapping json>'``
```
Comment (emilyliu7321) [inline README.rst:26]: Can't comment on the lines of code right below this, but the filepaths for fixtures are out-of-date now

File: src/elexclarity/formatters/results.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        the vote types accordingly.
+        """
+        subunits = defaultdict(lambda: 0)
+        clarity_level = level.capitalize()
+
+        for vote_type in get_list(choice.get("VoteType", [])):
+            vote_type_subunits = vote_type.get(clarity_level, [])
+            for vote_type_subunit in get_list(vote_type_subunits):
+                if level == "precinct":
+                    subunit_id = self.get_precinct_id(vote_type_subunit["name"])
```
Comment (emilyliu7321) [inline src/elexclarity/formatters/results.py:68]: Do we no longer want the precinct ids to have the county appended to the front?

Comment (st421) [reply src/elexclarity/formatters/results.py:68]: I don't think so. I was originally thinking we'd have a statewide results object with precinct subunits (in which case we'd want county + precinct IDs), but if the county results are the top-level import, then the data importer is going to automatically take that top level ID and use it with the subunit IDs to make something like <date>_GA_G_S_<county fips>_<precinct slug> (thus my making sure we've got the county FIPS in the top level ID)

File: src/elexclarity/client.py [hunk 4]
```diff
@@ -92,29 +88,9 @@ def get_settings(self, electionid, statepostal, **kwargs):
...
-        counties = []
-        for raw_county in raw_counties:
-            name, _id, version, last_updated = raw_county.split("|")[0:4]
-            counties.append(CountyIdentifier(_id, name, version, last_updated))
-        return counties
-
     # https://results.enr.clarityelections.com//GA//105369/270988/reports/detailxml.zip
-    def get_county_results(self, statepostal, county, **kwargs):
-        return self.get_detail_xml(f"{statepostal}/{county.name}/{county.id}/{county.version}")
+    def get_county_results(self, statepostal, county_name, county_id, county_version, **kwargs):
```
Comment (pandringa) [inline src/elexclarity/client.py:92]: Tiny pet peeve across this repo: we use `statepostal` everywhere, but in our other repos we usually use `state`. (This just forces me to copy one key into the other when preparing the fetcher params, so it's not the end of the world.)

The same thing happens for `electionid` / `election_id` below — it might be _slightly_ easier to standardize across all the repos, but isn't the end of the world either

**Review by emilyliu7321 — APPROVED** [review]
